libmain/src/main.cpp

```diff
@@ -1,24 +1,90 @@
-#include <cvs/common/configuration.hpp>
+#include <boost/program_options.hpp>
+#include <cvs/common/config.hpp>
 #include <cvs/common/factory.hpp>
+#include <cvs/logger/logging.hpp>
+#include <cvs/pipeline/imodulemanager.hpp>
 #include <cvs/pipeline/ipipeline.hpp>
+#include <cvs/pipeline/registrationhelper.hpp>
+#include <cvs/pipeline/tbb/tbbhelpers.hpp>
 
-DEFINE_TYPE(IPipeline, DefaultPipeline);
+#include <iostream>
+
+extern cvs::common::FactoryPtr<std::string> pipelineFactory();
+
+using namespace std::string_literals;
+namespace po = boost::program_options;
 
 int main(int argc, char *argv[]) {
-  auto confuguration = std::make_shared<cvs::common::Configuration>();
-  confuguration->init({});  // TODO: path to config from args
+  int result = 1;
 
-  std::string pipelinte_name = IPipelineDefaultPipelineKey;  // TODO: from confuguration ???
+  try {
+    std::string             module_manager_key;
+    std::string             pipeline_key;
+    std::string             config_path_string;
+    po::options_description desc("Allowed options:");
+    desc.add_options()                                                                                           //
+        ("help,h", "produce help message")                                                                       //
+        ("version,v", "CVSPipeline library version")                                                             //
+        ("config,c", po::value(&config_path_string)->default_value(DEFAULT_CONFIG), "Config file path")          //
+        ("pipeline,p", po::value(&pipeline_key)->default_value("Default"), "Key for pipeline object")            //
+        ("module,m", po::value(&module_manager_key)->default_value("Default"), "Key for module manager object")  //
+        ;                                                                                                        //
 
-  auto pipeline = cvs::common::Factory::create<cvs::pipeline::IPipeline>(pipelinte_name, confuguration);
-  if (!pipeline) {
-    // TODO: log error
-    return 1;
-  }
+    po::variables_map cmd_line_vars;
+    po::store(po::parse_command_line(argc, argv, desc), cmd_line_vars);
+    po::notify(cmd_line_vars);
+
+    if (cmd_line_vars.contains("help")) {
+      std::cout << desc << std::endl;
+      return 1;
```

Comment:
Программа должна успешно завершаться
```suggestion
      return 0;
```